kai_solution_server/service/incident_store/incident_store.py

```diff
@@ -447,7 +454,9 @@ def post_process(self, limit: int = 5) -> None:
         page_size = 100
         page = 0
         processed_count = 0
-        limit = float("inf") if limit < 0 else limit
+        # I have no idea what this is trying to do.
+        # we need a better way.
```

Comment:
It's making it so that processed_count will never go above limit if you set limit to a negative number, ie -1 -> infinity. I think. Never seen it before tbh

Reply:
yeah, i think, but the types are different and that messes things up.

Reply:
See my latest commit. It would require changing the while loop but that requires brain power right now (and potentially modifying the functionality)